UEFI File getInfo() method broken

### Zig Version

0.15.0-dev.1034+bd97b6618

### Steps to Reproduce and Observed Behavior

Just use the method. Here is my example:

```zig
const std = @import("std");
const uefi = std.os.uefi;

pub fn main() uefi.Status {
    var file_system: ?*uefi.protocol.SimpleFileSystem = null;
    _ = uefi.system_table.boot_services.?.locateProtocol(
        &uefi.protocol.SimpleFileSystem.guid,
        null,
        @ptrCast(&file_system),
    );

    const root_dir = file_system.?.openVolume() catch {
        return uefi.Status.aborted;
    };

    const file_size = root_dir.getInfoSize(uefi.protocol.File.Info.file) catch {
        return uefi.Status.aborted;
    };


    var info_buf_ptr: [*]align(8) u8 = undefined;
    const result = uefi.system_table.boot_services.?.allocatePool(
        .loader_data,
        file_size,
        &info_buf_ptr,
    );

    if (result != uefi.Status.success) {
        return uefi.Status.aborted;
    }


    const info_buf: []align(8) u8 = info_buf_ptr[0..83];
    _ = root_dir.getInfo(
        uefi.protocol.File.Info.file_system,
        info_buf,
    ) catch {
        return uefi.Status.aborted;
    };

    return uefi.Status.success;
}
```

Any way I use the `getInfo()` method, I always get the same error during compilation:

```
$ zig build
install
└─ install main
   └─ compile exe main Debug x86_64-uefi-msvc 1 errors
/usr/lib/zig/lib/std/os/uefi/protocol/file.zig:228:47: error: @ptrCast increases pointer alignment
            .success => return @as(*InfoType, @ptrCast(buffer.ptr)),
                                              ^~~~~~~~~~~~~~~~~~~~
/usr/lib/zig/lib/std/os/uefi/protocol/file.zig:228:62: note: '[*]u8' has alignment '1'
            .success => return @as(*InfoType, @ptrCast(buffer.ptr)),
                                                       ~~~~~~^~~~
/usr/lib/zig/lib/std/os/uefi/protocol/file.zig:228:47: note: '*os.uefi.protocol.file.File.Info.FileSystem' has alignment '8'
/usr/lib/zig/lib/std/os/uefi/protocol/file.zig:228:47: note: use @alignCast to assert pointer alignment
referenced by:
    main: src/main.zig:129:29
    EfiMain: /usr/lib/zig/lib/std/start.zig:219:42
    3 reference(s) hidden; use '-freference-trace=5' to see all references
```

### Expected Behavior

I would assume the method would be able to be used using the arguments I did pass.
I always tried to use the `align(8)` directive to the type, but the `getInfo()` buffer parameter doesn't have any kind of restrictions regarding alignment.

--- a/lib/std/os/uefi/protocol/file.zig
+++ b/lib/std/os/uefi/protocol/file.zig
@@ -225,7 +225,7 @@ pub const File = extern struct {
             &len,
             buffer.ptr,
         )) {
-            .success => return @as(*InfoType, @ptrCast(buffer.ptr)),
+            .success => return @as(*InfoType, @ptrCast(@alignCast(buffer.ptr))),
             .buffer_too_small => return Error.BufferTooSmall,
             .unsupported => return Error.Unsupported,
             .no_media => return Error.NoMedia,

--- a/lib/std/os/uefi/protocol/file.zig
+++ b/lib/std/os/uefi/protocol/file.zig
@@ -214,7 +214,7 @@ pub const File = extern struct {
     pub fn getInfo(
         self: *const File,
         comptime info: std.meta.Tag(Info),
-        buffer: []u8,
+        buffer: []align(@alignOf(@FieldType(Info, @tagName(info)))) u8,
     ) GetInfoError!*@FieldType(Info, @tagName(info)) {
         const InfoType = @FieldType(Info, @tagName(info));
 
@@ -225,7 +225,7 @@ pub const File = extern struct {
             &len,
             buffer.ptr,
         )) {
-            .success => return @as(*InfoType, @ptrCast(@alignCast(buffer.ptr))),
+            .success => return @as(*InfoType, @ptrCast(buffer.ptr)),
             .buffer_too_small => return Error.BufferTooSmall,
             .unsupported => return Error.Unsupported,
             .no_media => return Error.NoMedia,

